### PR TITLE
[mpfr] Add mirror for mpfr at gnu.org

### PR DESCRIPTION
**Pull request by michaelkonecny**
adding a download mirror after having experienced and extended downtime of mpfr.org

### CODE REVIEW

File: ports/mpfr/portfile.cmake [hunk 1]
```diff
@@ -1,7 +1,7 @@
 include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/mpfr/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/mpfr/portfile.cmake [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
 vcpkg_download_distfile(ARCHIVE
-    URLS "http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.xz"
+    URLS "http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.xz" "https://ftp.gnu.org/gnu/mpfr/mpfr-4.0.2.tar.xz"
     FILENAME "mpfr-4.0.2.tar.xz"
     SHA512 d583555d08863bf36c89b289ae26bae353d9a31f08ee3894520992d2c26e5683c4c9c193d7ad139632f71c0a476d85ea76182702a98bf08dde7b6f65a54f8b88
 )
```
Comment (LilyWangL) [inline ports/mpfr/portfile.cmake]: Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Comment by LilyWangL** [comment]
@michaelkonecny Can you please correct the Version field in the CONTROL file?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by michaelkonecny** [comment]
> @michaelkonecny Can you please correct the Version field in the CONTROL file?

This is my first time working with vcpkg. My though is maybe the "-1" in the version number is there for a reason. I don't want to break anything.

Are you sure it's ok to change it?

**Comment by LilyWangL** [comment]
You can update `Version: 4.0.2-1` to `Version: 4.0.2-2`. This is [maintainer-guide.md](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md) documention. You can get more information from it.

**Comment by LilyWangL** [comment]
Pinging @michaelkonecny for response. Is work still being done for this PR?

**Comment by michaelkonecny** [comment]
Hi, sorry, got distracted. I will do this tonight.

File: ports/mpfr/portfile.cmake (hunk 2 above)
Comment (michaelkonecny) [reply ports/mpfr/portfile.cmake]: 1) are you sure it is ok to have COPY and RENAME on the same line?
2) there is no LICENSE.md file in ${SOURCE_PATH}
3) I haven't found what ${PORT} is

Do you just want me to change COPY to INSTALL?
Please be more clear next time.

Comment (LilyWangL) [reply ports/mpfr/portfile.cmake]: Sorry, it is my fault. The command should be `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`. For this port, `${PORT}` is mpfr.

Comment (michaelkonecny) [reply ports/mpfr/portfile.cmake]: OK, thanks.

Done.

**Review by LilyWangL — APPROVED** [review]

**Comment by dan-shaw** [comment]
thanks for the PR!

**Pull request merged**